extensions/cactus-plugin-htlc-eth-besu/src/main/contracts/contracts/HashTimeLock.sol

```diff
@@ -0,0 +1,165 @@
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+contract HashTimeLock {
+
+    mapping(bytes32 => LockContract) public contracts;
+
+    //                   / - WITHDRAWN
+    // INVALID - ACTIVE |
+    //                   \ - EXPIRED - REFUNDED
+
+    uint256 public constant INVALID = 0; // Uninitialized  swap -> can go to ACTIVE
+    uint256 public constant ACTIVE = 1; // Active swap -> can go to WITHDRAWN or EXPIRED
+    uint256 public constant REFUNDED = 2; // Swap is refunded -> final state.
+    uint256 public constant WITHDRAWN = 3; // Swap is withdrawn -> final state.
+    uint256 public constant EXPIRED = 4; // Swap is expired -> can go to REFUNDED
+
+    struct LockContract {
+        uint256 inputAmount;
+        uint256 outputAmount;
+        uint256 expiration;
+        uint256 status;
+        bytes32 hashLock;
+        address payable sender;
+        address payable receiver;
+        string outputNetwork;
+        string outputAddress;
+    }
+
+    event Withdraw(
+        bytes32 indexed id,
+        bytes32 secret,
+        bytes32 hashLock,
+        address indexed sender,
+        address indexed receiver
+    );
+
+    event Refund(
+        bytes32 indexed id,
+        bytes32 hashLock,
+        address indexed sender,
+        address indexed receiver
+    );
+
+    event NewContract(
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint256 expiration,
+        bytes32 indexed id,
+        bytes32 hashLock,
+        address indexed sender,
+        address indexed receiver,
+        string outputNetwork,
+        string outputAddress
+    );
+
+    function newContract(
+        uint256 outputAmount,
+        uint256 expiration,
+        bytes32 hashLock,
+        address payable receiver,
+        string calldata outputNetwork,
+        string calldata outputAddress
+    ) external payable {
+        address payable sender = msg.sender;
+        uint256 inputAmount = msg.value;
```

Comment:
Did you assume gas is never charged? I thought gas for processing should be deducted here.